src/components/GraphiQL.js

```diff
@@ -678,7 +689,11 @@ export class GraphiQL extends React.Component {
         operationName,
         ...queryFacts
       });
+      
+      return operationName;
```

Comment:
What if instead of calling `setState` directly, this function *returned* the object. That way you could replace calls to `this._updateQueryFacts(query)` with `this.setState(this._updateQueryFacts(query))`, however if you expect that multiple setState calls will occur in a row, then you can merge them before calling setState.

Another root issue seems to be that this function uses `this.state` as the source of truth, however clearly there are cases where `operationName` should not be read from `this.state` and instead should be passed into the function. I would suggest accepting a 2nd argument for operationName and using that instead of `this.state. operationName` within this function.